core/src/smartcontracts/wasm.rs

```diff
@@ -629,111 +726,79 @@ impl<'wrld> Runtime<'wrld> {
             .map_err(|err| Error::Decode(err.into()))
     }
 
-    /// Executes the given wasm smartcontract
-    ///
-    /// # Errors
-    ///
-    /// - if unable to construct wasm module or instance of wasm module
-    /// - if unable to find expected main function export
-    /// - if the execution of the smartcontract fails
-    pub fn execute(
-        &mut self,
-        wsv: &mut WorldStateView,
-        authority: AccountId,
-        bytes: impl AsRef<[u8]>,
-    ) -> Result<()> {
-        let span = wasm_log_span!("Smart contract execution", %authority);
-        let state = State::new(wsv, authority, self.config, span);
-
-        self.execute_smart_contract_with_state(bytes, state)
-    }
-
-    fn execute_smart_contract_with_state(
-        &mut self,
-        bytes: impl AsRef<[u8]>,
-        state: State,
-    ) -> Result<()> {
-        let mut store = self.create_store(state);
-        let smart_contract = self.create_smart_contract(&mut store, bytes)?;
+    #[allow(clippy::needless_pass_by_value)]
+    #[codec::wrap]
+    fn execute_query_as_validator(
+        query: QueryBox,
+        state: &mut state::Validator,
+    ) -> Result<Value, ValidationFail> {
+        iroha_logger::debug!(%query, "Executing as validator");
 
-        Self::execute_main_with_store(&smart_contract, &mut store)
+        query.execute(state.common_mut().wsv).map_err(Into::into)
     }
 
-    fn execute_main_with_store(
-        instance: &wasmtime::Instance,
-        store: &mut wasmtime::Store<State>,
-    ) -> Result<()> {
-        let main_fn = Self::get_typed_func(instance, store, export::WASM_MAIN_FN_NAME)?;
+    #[codec::wrap]
+    fn execute_instruction_as_validator(
```

Comment:
```suggestion
    fn execute_instruction(
```

it's implicit in the type

Reply:
See the response for `execute_query` above